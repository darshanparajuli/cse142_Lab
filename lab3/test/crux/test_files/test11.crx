// test large-ish symbol table, with multiple errors

var gorn : void;
func main(a:void) : int {
  var b:float;
  let a = 1;
}
var kirk : boolean;
func destroy_all_humans(lambda: void, laMbda: int) : void
{
    var greek : int;
    while (not 4) {
        return 88 * ::main();
        var x : int;
        let _ = 8 and 9;
    }
    let x = 9;
    let y = 20;
}

var gorn : void;
