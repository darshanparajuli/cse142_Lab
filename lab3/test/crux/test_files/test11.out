Error parsing file.
ResolveSymbolError(15,13)[Could not find _.]
Symbol(readInt)
Symbol(readFloat)
Symbol(printBool)
Symbol(printInt)
Symbol(printFloat)
Symbol(println)
Symbol(gorn)
Symbol(main)
Symbol(kirk)
Symbol(destroy_all_humans)
  Symbol(lambda)
  Symbol(laMbda)
  Symbol(greek)
    Symbol(x)

ResolveSymbolError(17,9)[Could not find x.]
Symbol(readInt)
Symbol(readFloat)
Symbol(printBool)
Symbol(printInt)
Symbol(printFloat)
Symbol(println)
Symbol(gorn)
Symbol(main)
Symbol(kirk)
Symbol(destroy_all_humans)
  Symbol(lambda)
  Symbol(laMbda)
  Symbol(greek)

ResolveSymbolError(18,9)[Could not find y.]
Symbol(readInt)
Symbol(readFloat)
Symbol(printBool)
Symbol(printInt)
Symbol(printFloat)
Symbol(println)
Symbol(gorn)
Symbol(main)
Symbol(kirk)
Symbol(destroy_all_humans)
  Symbol(lambda)
  Symbol(laMbda)
  Symbol(greek)

DeclareSymbolError(21,5)[gorn already exists.]
Symbol(readInt)
Symbol(readFloat)
Symbol(printBool)
Symbol(printInt)
Symbol(printFloat)
Symbol(println)
Symbol(gorn)
Symbol(main)
Symbol(kirk)
Symbol(destroy_all_humans)


